runtime/src/chpl-gpu.c

```diff
@@ -87,6 +87,7 @@ static void override_number_of_devices(void) {
 }
 
 static void find_and_setup_devices(int numAllDevices) {
+#ifndef GPU_RUNTIME_CPU
```

Comment:
I don't think you'd disagree, but at a high-level, it'd be good to consider making the functions in the body of this ifndef "work" with the CPU as device mode. I mostly wanted to note, and not necessarily advocating that you spend time on this.

Reply:
The difficulty is that the main body is operating on topological info gather by hwloc -- but no such info would exist for "virtual" devices like made up GPUs. So, "gather topo info" does nothing, and passing this uninitialized data to the (non-GPU) device assignment code seems dangerous.

It would be nice to be able to simulate different configurations of simulated CPUs (e.g., such that we could pretend that we can determine different PCI bus IDs for different GPUs). But that's definitely beyond the factor, and dubiously worth the effort.

Reply:
> o, "gather topo info" does nothing, and passing this uninitialized data to the (non-GPU) device assignment code seems dangerous.

Oh, I see. Even if we were to make it work, it would require putting these `ifndef`s in those functions. That's worse, I think. Thanks for that context!